### PR TITLE
Update ac-smallmultiple.mdx

### PR DESCRIPTION
**Pull request by heisenberg183**
Adding Approach Section to this solution

_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/ac-smallmultiple.mdx [hunk 1]
```diff
@@ -15,6 +15,39 @@ author: Chongtian Ma
 
 **Time Complexity:** $\mathcal{O}(K)$
 
+## Aproach
```
Comment (SansPapyrus683) [inline solutions/gold/ac-smallmultiple.mdx:18]: put this above the implementation header and below the official analysis section

**Comment by SansPapyrus683** [comment]
@envyaims your problem, what you think of this

File: solutions/gold/ac-smallmultiple.mdx (hunk 1 above)
Comment (heisenberg183) [reply solutions/gold/ac-smallmultiple.mdx:10]: Done

File: solutions/gold/ac-smallmultiple.mdx [hunk 2]
```diff
@@ -7,6 +7,39 @@ author: Chongtian Ma
...
+are drawn, you can see that the vertices corresponding to each positive integer
+can be equated with modK.
+
+Therefore, we only need to find the shortest path from 1 to 0 on a graph of K
+vertices, where each vertex is equated with modK, and this can be found in O(K)
+time using 01BFS.
+
+A transition that adds 1 to an integer with a 1's digit of 9 does not need to be
+treated specially because if you recall the shortest path algorithm, you can see
+that the destination vertex has already been visited.
```
Comment (envyaims) [inline solutions/gold/ac-smallmultiple.mdx:36]: elaborate on this, how does this relate to the problem?

File: solutions/gold/ac-smallmultiple.mdx [hunk 3]
```diff
@@ -7,6 +7,39 @@ author: Chongtian Ma
 
 [Official Analysis](https://img.atcoder.jp/arc084/editorial.pdf)
 
+## Aproach
+
+All positive integers can be created by starting from 1 and repeating the
+following two operations.
+
+- Add 1 to the current number. At this time, the sum of each digit increases
```
Comment (envyaims) [inline solutions/gold/ac-smallmultiple.mdx:15]: swap the order of these two

**Review by envyaims — CHANGES_REQUESTED** [review]
looks great, thanks!

File: solutions/gold/ac-smallmultiple.mdx (hunk 3 above)
Comment (heisenberg183) [reply solutions/gold/ac-smallmultiple.mdx:17]: > swap the order of these two

Done

File: solutions/gold/ac-smallmultiple.mdx (hunk 2 above)
Comment (heisenberg183) [reply solutions/gold/ac-smallmultiple.mdx:37]: Done

**Review by envyaims — APPROVED** [review]
explanation looks fine, ill leave it to sanspapyrus to complain about formatting and latexing

File: solutions/gold/ac-smallmultiple.mdx [hunk 4]
```diff
@@ -7,6 +7,41 @@ author: Chongtian Ma
...
+can be equated with modK.
+
+Therefore, we only need to find the shortest path from 1 to 0 on a graph of K
+vertices, where each vertex is equated with modK, and this can be found in O(K)
+time using 01BFS.
+
+A transition that adds 1 to an integer with a 1's digit of 9 does not need to be
+treated specially because if you recall the shortest path algorithm, you can see
+that the destination vertex has already been visited.
+In reference to this problem here if we go from 9 to 10 we dont need to worry as vertex 10 will already be visited thus we can skip this case
```
Comment (envyaims) [inline solutions/gold/ac-smallmultiple.mdx:38]:
```suggestion
For example, if we go from 9 to 10 using the second case don't need to worry as vertex 10 will already be visited from the first case. 
```

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**